tests/v2/validation/rbac/globalrolesv2/namespacedrules_test.go

```diff
@@ -0,0 +1,446 @@
+	require.NoError(ns.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(ns.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(ns.T(), err)
+
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+
+	log.Info("Update the global role and remove a namespace.")
+	namespacedRules = map[string][]rbacv1.PolicyRule{
+		readPodsNS: {
+			readServiceAccountPolicy,
+		},
+	}
+
+	deleteANamespace := v3.GlobalRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: createdGlobalRole.Name,
+		},
+		NamespacedRules: namespacedRules,
+	}
+
+	_, err = rbac.UpdateGlobalRole(ns.client, &deleteANamespace)
+	require.NoError(ns.T(), err)
+
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+
+	log.Info("Verify user can list pods in the namespace from the namespaced rules")
+	userClient, err := ns.client.AsUser(createdUser)
+	require.NoError(ns.T(), err)
+	listPodsAsUser, err := userClient.WranglerContext.Core.Pod().List(readPodsNS, metav1.ListOptions{})
+	require.NoError(ns.T(), err)
+	listPodsAsAdmin, err := ns.client.WranglerContext.Core.Pod().List(readPodsNS, metav1.ListOptions{})
+	require.NoError(ns.T(), err)
+	
+	require.Equal(ns.T(), len(listPodsAsAdmin.Items), len(listPodsAsUser.Items))
+	require.Equal(ns.T(), listPodsAsAdmin.Items, listPodsAsUser.Items)
+
+	log.Info("Verify user cannot list crtbs in the deleted namespace from the namespaced rules")
+	_, err = userClient.WranglerContext.Mgmt.ClusterRoleTemplateBinding().List(readCrtbsNS, metav1.ListOptions{})
+	require.Error(ns.T(), err)
+	k8sError.IsForbidden(err)
+}
+
+func (ns *NamespacedRulesTestSuite) TestDeleteUserDeletesRolebindingsForNamespacedRules() {
+
+	subSession := ns.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with multiple namespacedRules.")
+
+	readPodsNS, err := createProjectAndAddANamespace(ns.client, "custom-sa-ns")
+	require.NoError(ns.T(),err)
+
+	namespacedRules := map[string][]rbacv1.PolicyRule{
+		readPodsNS: {
+			readServiceAccountPolicy,
+		},
+	}
+	createdGlobalRole, err := createGlobalRoleWithNamespacedRules(ns.client, namespacedRules)
+	require.NoError(ns.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(ns.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(ns.T(), err)
+
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+	
+	grbOwner, err := getGlobalRoleBindingForUser(ns.client, createdUser.ID)
+	require.NoError(ns.T(), err)
+	nameSelector := fmt.Sprintf("metadata.name=%s-%s", grbOwner, namespace)
+
+	err = ns.client.WranglerContext.Mgmt.User().Delete(createdUser.ID, &metav1.DeleteOptions{})
+	require.NoError(ns.T(), err)
+
+	rbs, err := rbac.ListRoleBindings(ns.client, localcluster, namespace, metav1.ListOptions{FieldSelector: nameSelector})
```

Comment:
small nit, could you add a log.Info statement before listing the role bindings that just says you're verifying the namespaced rules role bindings are deleted after deleting the user? I think it would help readability.